src/openforms/formio/utils.py

```diff
@@ -35,6 +35,64 @@ def iter_components(
             )
 
 
+def iterate_data_with_components(
```

Comment:
let's not put this in the utils and create the impression it is public API intended to be used in multiple places. it does the job for the bug that we're dealing with, but it should really be just an implementation detail because the code necessary for this upload handling is not very robust or elegant.

So, let's make it a private function by renaming it to `_iterate_data_with_components` and move it back into the `src/openforms/submissions/attachments.py` module